### PR TITLE
feat: update langchaingo to latest and add gemini

### LINKED ISSUE #739: add gemini support in arcadia
We should be able to use gemini apis in our platform:
- add a third party model service with new provider gemini

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it

go.mod update is traced by https://github.com/kubeagi/langchaingo/pull/3
1. update to tmc/langchaingo latest(current is [tmc/langchaingo@22159ce](https://github.com/tmc/langchaingo/tree/22159ce9c04f792792288ebb31ace276f581678b))
2. add https://github.com/tmc/langchaingo/pull/617
3. add https://github.com/tmc/langchaingo/pull/639
we are behind 233 commits: https://github.com/kubeagi/langchaingo/compare/dev...tmc:langchaingo:main

There are 2 main upstream feat changes: refactoring the LLM interface in preparation for **multi-modal models**, and the addition of the **Google AI model gemini**, which we can use in the github action test.
And many bugfix. 

#### Which issue(s) this PR fixes

Fixes #739 
Close #745

### CODE REVIEW

File: pkg/llms/llms.go [hunk 1]
```diff
@@ -29,10 +29,14 @@ const (
...
 	ZhiPuAI   LLMType = "zhipuai"
 	DashScope LLMType = "dashscope"
+	Gemini    LLMType = "gemini"
 	Unknown   LLMType = "unknown"
 )
 
-var OpenAIModels = []string{"gpt-3.5", "gpt-3.5-turbo"}
+var (
+	OpenAIModels = []string{"gpt-3.5", "gpt-3.5-turbo"}
+	GeminiModels = []string{"gemini-pro"}
```
Comment (bjwswang) [inline pkg/llms/llms.go:38]: Should be 
- gemini-1.0-pro
-  gemini-1.0-pro-vision
See https://cloud.google.com/vertex-ai/docs/generative-ai/learn/models#gemini-models

Besides  google's generative-ai-go has changed the name too https://github.com/google/generative-ai-go/commit/223b9f68d6fd478fd79b395478d7ea799f4b9d94

Comment (bjwswang) [reply pkg/llms/llms.go:38]: For `gemini-1.0-pro-vision` ,I am  not sure if this is allowed.

Comment (Abirdcfly) [reply pkg/llms/llms.go:38]: models/gemini-1.0-pro The best model for scaling across a wide range of tasks
models/gemini-1.0-pro-001 The best model for scaling across a wide range of tasks. This is a stable model that supports tuning.
models/gemini-1.0-pro-latest The best model for scaling across a wide range of tasks. This is the latest model.
models/gemini-1.0-pro-vision-latest The best image understanding model to handle a broad range of applications
models/gemini-pro The best model for scaling across a wide range of tasks
models/gemini-pro-vision The best image understanding model to handle a broad range of applications
models/embedding-001 Obtain a distributed representation of a text.

by https://pkg.go.dev/github.com/google/generative-ai-go/genai#Client.ListModels

Comment (Abirdcfly) [reply pkg/llms/llms.go:38]: And "Note: gemini-pro is an alias for gemini-1.0-pro." in https://ai.google.dev/models/gemini

File: config/samples/app_shared_llm_service_gemini.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: app-shared-llm-secret
+  namespace: arcadia
+type: Opaque
+data:
+  apiKey: "QUl6YVN5QVZOdGRYOHpkeU5pNWpubzNYSExUWGM0UnpJSGxIRUFz"
```
Comment (nkwangleiGIT) [inline config/samples/app_shared_llm_service_gemini.yaml:8]: is any security risk here?

File: pkg/appruntime/log/log.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2023 KubeAGI.
```
Comment (bjwswang) [inline pkg/appruntime/log/log.go:2]: 2024

**Review by bjwswang — APPROVED** [review]

**Comment by bjwswang** [comment]
@Abirdcfly  Next we should consider https://github.com/kubeagi/arcadia/issues/779

**Pull request merged**